test/framefusion.test.ts

```diff
@@ -63,431 +45,180 @@ describe('framefusion', () => {
-                expect(Math.floor(extractor.ptsToTime(frame.pts) * 30)).to.equal(i);
-            }
-
-            // Cleanup
-            extractor.dispose();
-        }, 10000);
-
-        it.concurrent('Should seek and dump all frames in the video where the stream index is 1', async() => {
-            // Arrange
-            const extractor = await BeamcoderExtractor.create({
-                inputFileOrUrl: './test/samples/audio-is-stream0-video-is-stream1.mp4',
-                threadCount: 8,
-            }) as BeamcoderExtractor;
-
-            // Sample at the middle of each frame
-            const FRAME_SYNC_DELTA = (1 / 30.0) / 2.0;
-
-            // Act & assert
-            for (let i = 0; i < 60; i++) {
-                const frame = await extractor.getFrameAtTime(i / 30.0 + FRAME_SYNC_DELTA);
-                expect(Math.floor(extractor.ptsToTime(frame.pts) * 30)).to.equal(i);
-            }
-
-            // Cleanup
-            extractor.dispose();
-        }, 10000);
-
-        it('Should get frame as image data', async() => {
-            // Arrange
-            const extractor = await BeamcoderExtractor.create({
-                inputFileOrUrl: './test/samples/countTo60.mp4',
-                threadCount: 8,
-            }) as BeamcoderExtractor;
-
-            // Sample at the middle of each frame
-            const FRAME_SYNC_DELTA = (1 / 30.0) / 2.0;
-
-            // Act & assert
-            for (let i = 0; i < 10; i++) {
-                const imagedata = await extractor.getImageDataAtTime(i / 30.0 + FRAME_SYNC_DELTA);
-                expect(imagedata.width).to.equal(extractor.width);
-                expect(imagedata.height).to.equal(extractor.height);
-                const canvas = createCanvas(imagedata.width, imagedata.height);
-                const ctx = canvas.getContext('2d');
-                ctx.putImageData(imagedata, 0, 0);
-                expect(canvas.toBuffer('image/png')).toMatchImageSnapshot();
-            }
-
-            // Cleanup
-            extractor.dispose();
-        }, 2000);
-
-        it.concurrent('Should seek and dump all frames in the video [other video]', async() => {
-            // Arrange
-            const extractor = await BeamcoderExtractor.create({
-                inputFileOrUrl: TEST_VIDEO_LOW_FRAMERATE,
-                threadCount: 8,
-            }) as BeamcoderExtractor;
-
-            // Sample at the middle of each frame
-            const FRAME_SYNC_DELTA = (1 / 30.0) / 2.0;
-
-            // Act & assert
-            for (let i = 0; i < extractor.duration * 30.0; i++) {
-                const frame = await extractor.getFrameAtTime(i / 30.0 + FRAME_SYNC_DELTA);
-                expect(Math.floor(extractor.ptsToTime(frame.pts) * 30)).to.be.closeTo(i, 15);
-            }
-
-            // Cleanup
-            extractor.dispose();
-        }, 10000);
-
-        it.concurrent('Should seek and dump frames at different points in the video', async() => {
```

Comment:
This test also could help spot issues quickly in the future

Reply:
Actually no, an image test would be better than this test

Reply:
I think `can get frames at random times (forward and backward)` is what you're looking for